libraries/AP_ExternalAHRS/AP_ExternalAHRS_Scripting.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  support for scripting based external AHRS
+ */
+
+#include "AP_ExternalAHRS_config.h"
+
+#if AP_EXTERNAL_AHRS_SCRIPTING_ENABLED
+
+#include "AP_ExternalAHRS_Scripting.h"
+
+extern const AP_HAL::HAL &hal;
+
+// accessors for AP_AHRS
+bool AP_ExternalAHRS_Scripting::healthy(void) const
+{
+    WITH_SEMAPHORE(state.sem);
+    return AP_HAL::millis() - last_update_ms < 100;
+}
+
+bool AP_ExternalAHRS_Scripting::initialised(void) const
+{
+    WITH_SEMAPHORE(state.sem);
+    return last_update_ms != 0;
+}
+
+// handle input from scripting backend
+bool AP_ExternalAHRS_Scripting::handle_scripting(const AP_ExternalAHRS::state_t &_state, nav_filter_status_flags_t &_filter_status)
+{
+    WITH_SEMAPHORE(state.sem);
+
+    state.accel = _state.accel;
+    state.gyro = _state.gyro;
+
+    if (_state.have_quaternion) {
+        state.quat = _state.quat;
+    }
+
+    if (_state.have_location) {
+        state.location = _state.location;
+    }
+
+    if (_state.have_velocity) {
+        state.velocity = _state.velocity;
+    }
+
+    if (_state.have_origin && !state.have_origin) {
```

Comment:
This confused me for a few moments, that extra `_` is a bit subtle. Maybe `new_state` rater than `_state`?